components/brave_wallet/browser/swap_controller_unittest.cc

```diff
@@ -0,0 +1,243 @@
+      base::BindOnce(&OnRequestResponse, &callback_run, false, nullptr, error));
+  base::RunLoop().RunUntilIdle();
+  EXPECT_TRUE(callback_run);
+}
+
+TEST_F(SwapControllerUnitTest, GetTransactionPayload) {
+  SetInterceptor(
+      R"(
+      {
+        "price":"1916.27547998814058355",
+        "guaranteedPrice":"1935.438234788021989386",
+        "to":"0xdef1c0ded9bec7f1a1670819833240f027b25eff",
+        "data":"0x0",
+        "value":"0",
+        "gas":"719000",
+        "estimatedGas":"719000",
+        "gasPrice":"26000000000",
+        "protocolFee":"0",
+        "minimumProtocolFee":"0",
+        "buyTokenAddress":"0xeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee",
+        "sellTokenAddress":"0x6b175474e89094c44da98b954eedeac495271d0f",
+        "buyAmount":"1000000000000000000000",
+        "sellAmount":"1916275479988140583549706",
+        "sources":[],
+        "allowanceTarget":"0xdef1c0ded9bec7f1a1670819833240f027b25eff",
+        "sellTokenToEthRate":"1900.44962824532464391",
+        "buyTokenToEthRate":"1"
+      }
+    )");
+
+  auto expected_swap_response = brave_wallet::mojom::SwapResponse::New();
+  expected_swap_response->price = "1916.27547998814058355";
+  expected_swap_response->guaranteed_price = "1935.438234788021989386";
+  expected_swap_response->to = "0xdef1c0ded9bec7f1a1670819833240f027b25eff";
+  expected_swap_response->data = "0x0";
+  expected_swap_response->value = "0";
+  expected_swap_response->gas = "719000";
+  expected_swap_response->estimated_gas = "719000";
+  expected_swap_response->gas_price = "26000000000";
+  expected_swap_response->protocol_fee = "0";
+  expected_swap_response->minimum_protocol_fee = "0";
+  expected_swap_response->buy_token_address =
+      "0xeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee";
+  expected_swap_response->sell_token_address =
+      "0x6b175474e89094c44da98b954eedeac495271d0f";
+  expected_swap_response->buy_amount = "1000000000000000000000";
+  expected_swap_response->sell_amount = "1916275479988140583549706";
+  expected_swap_response->allowance_target =
+      "0xdef1c0ded9bec7f1a1670819833240f027b25eff";
+  expected_swap_response->sell_token_to_eth_rate = "1900.44962824532464391";
+  expected_swap_response->buy_token_to_eth_rate = "1";
+
+  bool callback_run = false;
+  swap_controller_->GetTransactionPayload(
+      GetCannedSwapParams(),
+      base::BindOnce(&OnRequestResponse, &callback_run, true,
+                     std::move(expected_swap_response), absl::nullopt));
+  base::RunLoop().RunUntilIdle();
+  EXPECT_TRUE(callback_run);
+}
+
+TEST_F(SwapControllerUnitTest, GetTransactionPayloadError) {
+  std::string error =
+      R"({"code":100,"reason":"Validation Failed","validationErrors":[{"field":"sellAmount","code":1000,"reason":"should have required property 'sellAmount'"},{"field":"buyAmount","code":1000,"reason":"should have required property 'buyAmount'"},{"field":"","code":1001,"reason":"should match exactly one schema in oneOf"}]})";
```

Comment:
I know this doesn't trigger lint failure because of R"" string but you can copy this string to vim
and do `%!python -m json.tool` and then you will get
```
{
    "code": 100,
    "reason": "Validation Failed",
    "validationErrors": [
        {
            "code": 1000,
            "field": "sellAmount",
            "reason": "should have required property 'sellAmount'"
        },
        {
            "code": 1000,
            "field": "buyAmount",
            "reason": "should have required property 'buyAmount'"
        },
        {
            "code": 1001,
            "field": "",
            "reason": "should match exactly one schema in oneOf"
        }
    ]
}
```
And you just need to add R"()"

Reply:
Fixed and rebased in f7c976cca65d7ac157387a7742807ec19430a3ce